datacapture/src/main/java/com/google/android/fhir/datacapture/fhirpath/FHIRPathEngineHostServices.kt

```diff
@@ -26,7 +26,8 @@ import org.hl7.fhir.r4.utils.FHIRPathEngine
  */
 internal object FHIRPathEngineHostServices : FHIRPathEngine.IEvaluationContext {
   override fun resolveConstant(appContext: Any?, name: String?, beforeContext: Boolean): Base? {
-    return if (appContext is Map<*, *> && appContext.containsKey(name)) appContext[name] as Base
+    return if (appContext is Map<*, *> && appContext.containsKey(name) && appContext[name] != null)
+      appContext[name] as Base
```

Comment:
optional : Can you please check below suggestion
```suggestion
     return (appContext as? Map<*, *>)?.get(name) as? Base
```

Reply:
safe casting should work in this case

Reply:
yep it works, I'll also remove the else statement